File: LIS.Web/src/app/models/billing-invoice-header.ts
export interface BillingInvoiceHeader {
  id: number;
  sequenceNumber: number;
  type: string;
  counterTypeId: number;
  counter: number;
  date: Date;
  admission: number;
  comment?: string;
  hospitalAmount: number;
  physicianAmount: number;
  medicamentAmount: number;
  discount: number;
  deniedAmount: number;
  lumpSum: number;
  accountId: number;
  accountDescription: string;
  complementaryAccountId?: number;
  complementaryAccountDescription?: string;
  currencyId: number;
  currency: string;
  exchangeRate: number;
  checkInClassId: number;
  checkInClass: string;
  coverageClassId: number;
  coverageClass: string;
  coverageRate: number;
  referralPhysicianId: number;
  referralPhysician: string;
  attendingPhysicianId?: number;
  attendingPhysician?: string;
  reference?: string;
  mainInvoice?: number;
  oldMainInvoice?: number;
  net: number;
  gross: number;
  netGross: number;
  complementary: number;
  complementaryOtherCurrency: number;
  complementaryDifferenceOtherCurrency: number;
  mrn: string;
  patientName: string;
  admissionNumber: string;
  admissionDate: Date;
  departmentId: number;
  department: string;
  dischargeDate?: Date;
  contextPriceId: number;
  contextPrice: string;
  receiptNumber?: string;
  receiptAmount?: number;
  receiptDate?: Date;
  roundedAmount?: number;
  insurance: number;
  admissionInsuranceCoverage: number;
  isDRG: number;
  collectionScheduleId?: number;
  collectionScheduleNumber?: string;
  collectionScheduleDate?: Date;
  receivedLBP?: number;
  receivedUSD?: number;
  difference?: number;
  receivingDate?: Date;
  voucherNumber?: string;
  splitedInvoice: number;
  primaryDischargeDiagnostic?: string;
  secondaryDischargeDiagnostic?: string;
  requireRegenerate: number;
  lockedBy?: number;
  lockedByName?: string;
  lockedDate?: Date;
  alternateInvoiceId?: number;
  globalDiscount: number;
  differenceAdjust?: number;
  modifiedDate: Date;
  isDeleted: number;
  createdBy: number;
  modifiedBy?: number;
  createdDate: Date;
  group?: number;
  agreementNumber?: string;
  complementaryDifferenceCalculationState?: number;
  isReversed: number;
  creditNoteNumber?: number;
  creditNoteDate?: Date;
  creditNotePaidAmount?: number;
  creditNoteDiscount?: number;
  employeeAccount?: string;
  isEmployee?: number;
  status?: number;
  contextEnumerationId?: number;
  isDirty?: number;
  isFromScratch?: number;
  agreementCreditAmount?: number;
  isSelected?: number;
  creditNoteAssignedAmount?: number;
  creditNoteVoucherNumber?: string;
  prepaymentAmount?: number;
  prepaymentDate?: Date;
  prepaymentNumber?: number;
  diagnosticGroup1?: string;
  diagnosticGroup2?: string;
  diagnosticGroup3?: string;
  diagnosticGroupId1?: number;
  diagnosticGroupId2?: number;
  diagnosticGroupId3?: number;
}

File: LIS.Api/Models/BillingInvoiceHeader.cs
using System.ComponentModel.DataAnnotations;

namespace LIS.Api.Models
{
    public class BillingInvoiceHeader
    {
        [Key]
        public int Id { get; set; }
        public int SequenceNumber { get; set; }
        public string Type { get; set; } = string.Empty;
        public int CounterTypeId { get; set; }
        public int Counter { get; set; }
        public DateTime Date { get; set; }
        public int Admission { get; set; }
        public string? Comment { get; set; }
        public decimal HospitalAmount { get; set; }
        public decimal PhysicianAmount { get; set; }
        public decimal MedicamentAmount { get; set; }
        public decimal Discount { get; set; }
        public decimal DeniedAmount { get; set; }
        public decimal LumpSum { get; set; }
        public int AccountId { get; set; }
        public string AccountDescription { get; set; } = string.Empty;
        public int? ComplementaryAccountId { get; set; }
        public string? ComplementaryAccountDescription { get; set; }
        public int CurrencyId { get; set; }
        public string Currency { get; set; } = string.Empty;
        public decimal ExchangeRate { get; set; }
        public int CheckInClassId { get; set; }
        public string CheckInClass { get; set; } = string.Empty;
        public int CoverageClassId { get; set; }
        public string CoverageClass { get; set; } = string.Empty;
        public decimal CoverageRate { get; set; }
        public int ReferralPhysicianId { get; set; }
        public string ReferralPhysician { get; set; } = string.Empty;
        public int? AttendingPhysicianId { get; set; }
        public string? AttendingPhysician { get; set; }
        public string? Reference { get; set; }
        public int? MainInvoice { get; set; }
        public int? OldMainInvoice { get; set; }
        public decimal Net { get; set; }
        public decimal Gross { get; set; }
        public decimal NetGross { get; set; }
        public decimal Complementary { get; set; }
        public decimal ComplementaryOtherCurrency { get; set; }
        public decimal ComplementaryDifferenceOtherCurrency { get; set; }
        public string MRN { get; set; } = string.Empty;
        public string PatientName { get; set; } = string.Empty;
        public string AdmissionNumber { get; set; } = string.Empty;
        public DateTime AdmissionDate { get; set; }
        public int DepartmentId { get; set; }
        public string Department { get; set; } = string.Empty;
        public DateTime? DischargeDate { get; set; }
        public int ContextPriceId { get; set; }
        public string ContextPrice { get; set; } = string.Empty;
        public string? ReceiptNumber { get; set; }
        public decimal? ReceiptAmount { get; set; }
        public DateTime? ReceiptDate { get; set; }
        public decimal? RoundedAmount { get; set; }
        public int Insurance { get; set; }
        public int AdmissionInsuranceCoverage { get; set; }
        public int IsDRG { get; set; }
        public int? CollectionScheduleId { get; set; }
        public string? CollectionScheduleNumber { get; set; }
        public DateTime? CollectionScheduleDate { get; set; }
        public decimal? ReceivedLBP { get; set; }
        public decimal? ReceivedUSD { get; set; }
        public decimal? Difference { get; set; }
        public DateTime? ReceivingDate { get; set; }
        public string? VoucherNumber { get; set; }
        public int SplitedInvoice { get; set; }
        public string? PrimaryDischargeDiagnostic { get; set; }
        public string? SecondaryDischargeDiagnostic { get; set; }
        public int RequireRegenerate { get; set; }
        public int? LockedBy { get; set; }
        public string? LockedByName { get; set; }
        public DateTime? LockedDate { get; set; }
        public int? AlternateInvoiceId { get; set; }
        public decimal GlobalDiscount { get; set; }
        public decimal? DifferenceAdjust { get; set; }
        public DateTime ModifiedDate { get; set; }
        public int IsDeleted { get; set; }
        public int CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public int? Group { get; set; }
        public string? AgreementNumber { get; set; }
        public int? ComplementaryDifferenceCalculationState { get; set; }
        public int IsReversed { get; set; }
        public int? CreditNoteNumber { get; set; }
        public DateTime? CreditNoteDate { get; set; }
        public decimal? CreditNotePaidAmount { get; set; }
        public decimal? CreditNoteDiscount { get; set; }
        public string? EmployeeAccount { get; set; }
        public int? IsEmployee { get; set; }
        public int? Status { get; set; }
        public int? ContextEnumerationId { get; set; }
        public int? IsDirty { get; set; }
        public int? IsFromScratch { get; set; }
        public decimal? AgreementCreditAmount { get; set; }
        public int? IsSelected { get; set; }
        public decimal? CreditNoteAssignedAmount { get; set; }
        public string? CreditNoteVoucherNumber { get; set; }
        public decimal? PrepaymentAmount { get; set; }
        public DateTime? PrepaymentDate { get; set; }
        public int? PrepaymentNumber { get; set; }
        public string? DiagnosticGroup1 { get; set; }
        public string? DiagnosticGroup2 { get; set; }
        public string? DiagnosticGroup3 { get; set; }
        public int? DiagnosticGroupId1 { get; set; }
        public int? DiagnosticGroupId2 { get; set; }
        public int? DiagnosticGroupId3 { get; set; }
    }
}